### PR TITLE
【hydra No.6】cylinder2d_unsteady

### PR DESCRIPTION
**Pull request by zbt78**
### PR types
Others

### PR changes
Others

### Describe
https://github.com/PaddlePaddle/PaddleScience/issues/571
No.9 cylinder2d_unsteady
1、使用第三方库 [hydra](https://hydra.cc/docs/0.11/intro/)适配 cylinder2d_unsteady案例
2、cylinder2d_unsteady案例代码重构，添加评估函数 evaluate，及其训练/评估命令和预训练模型、指标

### CODE REVIEW

File: docs/zh/examples/cylinder2d_unsteady.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
 
+=== "模型训练命令"
+
+    ``` sh
+    ```
+
+=== "模型评估命令"
+
+    ``` sh
+    ```
```
Comment (HydrogenSulfate) [inline docs/zh/examples/cylinder2d_unsteady.md:13]: 评估命令可以先删除

File: docs/zh/examples/cylinder2d_unsteady.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
 
 <a href="https://aistudio.baidu.com/aistudio/projectdetail/6160381?contributionType=1&sUid=438690&shared=1&ts=1683961158552" class="md-button md-button--primary" style>AI Studio快速体验</a>
 
+=== "模型训练命令"
+
+    ``` sh
```
Comment (HydrogenSulfate) [inline docs/zh/examples/cylinder2d_unsteady.md:7]:
``` sh
# linux
wget https://paddle-org.bj.bcebos.com/paddlescience/datasets/cylinder2d_unsteady_Re100/cylinder2d_unsteady_Re100_dataset.tar
# windows
# curl https://paddle-org.bj.bcebos.com/paddlescience/datasets/cylinder2d_unsteady_Re100/cylinder2d_unsteady_Re100_dataset.tar --output cylinder2d_unsteady_Re100_dataset.tar
# unzip it
tar -xvf bracket_dataset.tar
python cylinder2d_unsteady_Re100.py
```

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 3]
```diff
@@ -12,44 +12,42 @@
...
+    logger.init_logger("ppsci", osp.join(cfg.output_dir, "train.log"), "info")
 
     # set model
-    model = ppsci.arch.MLP(("t", "x", "y"), ("u", "v", "p"), 5, 50, "tanh")
+    model = ppsci.arch.MLP(**cfg.MODEL)
 
     # set equation
-    equation = {"NavierStokes": ppsci.equation.NavierStokes(0.02, 1.0, 2, True)}
+    equation = {
+        "NavierStokes": ppsci.equation.NavierStokes(cfg.viscosity, cfg.density, 2, True)
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:38]: ==> cfg.VISCOSITY, cfg.DENSITY

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 4]
```diff
@@ -81,13 +79,11 @@
...
     }
 
-    # set dataloader config
-    ITERS_PER_EPOCH = 1
-
     # pde/bc/sup constraint use t1~tn, initial constraint use t0
-    NPOINT_PDE, NTIME_PDE = 9420, len(train_timestamps)
-    NPOINT_INLET_CYLINDER = 161
-    NPOINT_OUTLET = 81
+    NPOINT_PDE = cfg.NPOINT_PDE
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:83]: 直接使用cfg.NPOINT_PDE代替NPOINT_PDE

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 5]
```diff
@@ -81,13 +79,11 @@
...
+    NTIME_PDE = len(train_timestamps)
+    NPOINT_INLET_CYLINDER = cfg.NPOINT_INLET_CYLINDER
+    NPOINT_OUTLET = cfg.NPOINT_OUTLET
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:86]: 同上

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 6]
```diff
@@ -214,17 +209,18 @@
...
         optimizer,
         None,
-        EPOCHS,
-        ITERS_PER_EPOCH,
-        eval_during_train=True,
-        eval_freq=EVAL_FREQ,
+        cfg.TRAIN.epochs,
+        cfg.TRAIN.iters_per_epoch,
+        eval_during_train=cfg.TRAIN.eval_during_train,
+        eval_freq=cfg.TRAIN.EVAL_FREQ,
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:218]: EVAL_FREQ==>eval_freq

File: examples/cylinder/2d_unsteady/conf/cylinder2d_unsteady.yaml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+# training settings
+TRAIN:
+  iters_per_epoch: 1
+  epochs: 40000
+  EVAL_FREQ: 400
+  learning_rate: 0.001
+  eval_during_train: true
+  checkpoint_path: null
+  pretrained_model_path: null
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/conf/cylinder2d_unsteady.yaml:64]: TRAIN.pretrained_model_path可以删除

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 8]
```diff
@@ -233,18 +229,121 @@
...
+    ppsci.utils.misc.set_random_seed(cfg.seed)
+
+    # initialize logger
+    logger.init_logger("ppsci", osp.join(cfg.output_dir, "eval.log"), "info")
+
+    # set model
+    model = ppsci.arch.MLP(**cfg.MODEL)
+
+    # set equation
+    equation = {"NavierStokes": ppsci.equation.NavierStokes(0.02, 1.0, 2, True)}
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:244]: 这里的方程仍然可以使用cfg的参数

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 9]
```diff
@@ -233,18 +229,121 @@
...
+            ),
+            ppsci.geometry.PointCloud(
+                reader.load_csv_file(
+                    "./datasets/domain_train.csv",
+                    ("x", "y"),
+                    alias_dict={"x": "Points:0", "y": "Points:1"},
+                ),
+                ("x", "y"),
+            ),
+        ),
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:277]: `time_rect`这个几何在评估函数中没有使用，可以删除

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 10]
```diff
@@ -233,18 +229,121 @@
...
+        "time_rect_eval": ppsci.geometry.PointCloud(
+            reader.load_csv_file(
+                "./datasets/domain_eval.csv",
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py:280]: 该路径使用cfg.DOMAIN_EVAL_PATH控制

File: examples/cylinder/2d_unsteady/conf/cylinder2d_unsteady.yaml [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    # output directory for multirun
+    dir: ${hydra.run.dir}
+    subdir: ./
+
+# set constraint data path
+DOMAIN_INLET_CYLINDER_PATH: ./datasets/domain_inlet_cylinder.csv
+DOMAIN_OUTLET_PATH: ./datasets/domain_outlet.csv
+IC0_1_PATH: ./datasets/initial/ic0.1.csv
+PROBE1_50_PATH: ./datasets/probe/probe1_50.csv
+
```
Comment (HydrogenSulfate) [inline examples/cylinder/2d_unsteady/conf/cylinder2d_unsteady.yaml:27]: 新增 `# set validator data path`，存放DOMAIN_EVAL_PATH

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py [hunk 12]
```diff
@@ -233,18 +229,121 @@
...
+        output_dir=cfg.output_dir,
         validator=validator,
         visualizer=visualizer,
-        pretrained_model_path=f"{OUTPUT_DIR}/checkpoints/latest",
+        pretrained_model_path=cfg.EVAL.pretrained_model_path,
     )
+    # evaluate after finished training
     solver.eval()
-    # visualize prediction from pretrained_model_path(optional)
+    # visualize prediction after finished training
```
[the remark this answers is not in the document]
Comment (zbt78) [reply examples/cylinder/2d_unsteady/cylinder2d_unsteady_Re100.py]: All Done.

**Review by HydrogenSulfate — APPROVED** [review]
LGTM
